### ISSUE TITLE
Confusing behavior of ignore-mode "download"

### ISSUE THREAD

**Issue by emanruse** [newsboat/newsboat#1361]
If `ignore-mode "download"` is set, the common sense expectation is non-visible articles not to exist in local cache. However, when set *after* the articles have already been downloaded, they remain.

This is not really a bug, but it deserves some attention. Possible solutions:

1. The downloaded (and invisible) articles should be deleted to match the expectation of a clean local cache (perhaps after a warning and confirmation after enabling the ignore-mode)

2. An additional option to deal properly with such situation. For example: `ignore-mode "delete"` which would do the same which "download" does plus also clean up local cache from old articles matching the killfiles.

Tested in:
```
 emanruse commented Dec 15, 2020

newsboat 2.21.0 - https://newsboat.org/
Copyright (C) 2006-2015 Andreas Krennmair
Copyright (C) 2015-2020 Alexander Batischev
Copyright (C) 2006-2017 Newsbeuter contributors
Copyright (C) 2017-2020 Newsboat contributors

Newsboat is free software licensed under the MIT License. (Type `newsboat -vv' to see the full text.)
It bundles:
- JSON for Modern C++ library, licensed under the MIT License: https://github.com/nlohmann/json
- optional-lite library, licensed under the Boost Software License: https://github.com/martinmoene/optional-lite
- expected-lite library, licensed under the Boost Software License: https://github.com/martinmoene/expected-lite

newsboat 2.21.0
System: Linux 5.9.12-1-default (i686)
Compiler: g++ 10.2.1 20200825 [revision c0746a1beb1ba073c7981eb09f55b3d993b32e5c]
ncurses: ncurses 6.2.20201031 (compiled with 6.2)
libcurl: libcurl/7.73.0 OpenSSL/1.1.1h-fips zlib/1.2.11 libidn2/2.3.0 libpsl/0.21.1 (+libidn2/2.3.0) libssh/0.9.5/openssl/zlib nghttp2/1.41.0 (compiled with 7.73.0)
SQLite: 3.34.0 (compiled with 3.33.0)
libxml2: compiled with 2.9.10
```

**Comment by Minoru** [comment]
You're right, this option could've been more intuitive. Out of the approaches you listed, I'd prefer the second one, because it's backwards-compatible and lets people choose between keeping old items and deleting them retroactively.

Actually, the more I think about it, the more I wonder why Newsboat even has `ignore-mode "display"`. Isn't it the same as filtering the feed? I'll have to dive into the Git history to see how this feature got developed, before I can decide what to do with it going further.

**Comment by emanruse** [comment]
> I'd prefer the second one

Agreed.

> Actually, the more I think about it, the more I wonder why Newsboat even has `ignore-mode "display"`. Isn't it the same as filtering the feed?

Yes.

**Comment by Minoru** [comment]
Okay, so `ignore-mode "display"` was added in Newsbeuter 2.2 via https://github.com/newsboat/newsboat/commit/1c8b9e440c438e0a51ee0657c72f2cb903a1cc39 to fix https://code.google.com/archive/p/newsbeuter/issues/52. The discussion in that issue draws a clear analogy with filters, and even suggests using pre-defined filters for that.

At the time, apparently no-one was inconvenienced by `ignore-mode "download"` not removing articles that are already in the cache.

I think it's fine to add `ignore-mode "delete"` (name subject to bikeshedding still), so I'm re-labelling this as an enhancement.

**Comment by emanruse** [comment]
Sounds good.
